File: boj/b_2133.js
function solution(n) {
  const table = Array(n + 1).fill(0);
  table[0] = 1;
  for (let i = 2; i <= n; i += 2) {
    table[i] += table[i - 2] * 3;
    for (let j = 0; j <= i - 4; j += 1) table[i] += table[j] * 2;
  }
  return table[n];
}
console.log(solution(6));

File: boj/b_2133.py
import sys
sys.stdin = open("input.txt", "r") # 제거
input = sys.stdin.readline

n = int(input())
table = [1] + [0] * n
for i in range(2, n + 1, 2):
    table[i] += table[i - 2] * 3
    for j in range(0, i - 2, 2):
        table[i] += table[j] * 2
print(table[n])